x/mongo/driver/operation.go

```diff
@@ -683,8 +688,7 @@ func (op Operation) Execute(ctx context.Context) error {
 			first = false
 		}
 
-		// Calculate maxTimeMS value to potentially be appended to the wire message.
-		maxTimeMS, err := op.calculateMaxTimeMS(ctx, srvr.RTTMonitor().P90(), srvr.RTTMonitor().Stats())
+		maxTimeMS, err := op.calculateMaxTimeMS(ctx, srvr.RTTMonitor())
```

Comment:
@matthewdale We just discovered an edge case here. We apparently have a background job that sets a context timeout of ~25 days, more specifically `2211839988` milliseconds which is greater than 32-bit max int so they started getting `BadValue) 2211839988 value for maxTimeMS is out of range [0, 2147483647]` Obviously this is a somewhat far off edge case, I'm asking the team what's the reason of having a context with a deadline this long.

But wanted to flag this to you since this PR is going to start erroring out on those edge cases. The patch we applied internally is to just take `min(math.MaxInt32 - (2 * time.Millisecond), time.Duration(float64(timeLeft)))` to avoid triggering that validation error. Not sure if the logic in `calculateMaxTimeMS` should just say "That timeout is too long, I'm not going to set maxTimeMS at all" or if we'd want to apply the same `min` logic.

Reply:
Ah the explanation of why the timeout is so long makes sense. tl;dr - Processes for things like refunds/chargebacks can have month long time frames, and various workflow orchestration systems will set a golang context to timeout when that window ends even if the current step of the workflow won't take nearly that long. 

Our team is going to look at tuning DB calls to have more granular contexts/timeouts, but this still might be a broader issue y'all might be concerned about since others could run into it

Reply:
Wow, nice find! I've updated the driver to omit `maxTimeMS` if the calculated value is greater than max int32 so that there's no timeout you can set that will cause the "BadValue" error.

That means if an application starts an op with a timeout greater than 2147483647ms, there will be no database-side timeout automatically set. However, the client-side timeout logic will still work and will cause the driver to close the connection if that timeout is ever reached, cancelling the op on the server-side (unless the command returns within the new 1-second "grace period" after the timeout).

Reply:
Awesome, that approach makes sense to me and the code looks good. Nice job on the quick fix!